src/python/pants/core/util_rules/external_tool.py

```diff
@@ -175,6 +226,29 @@ def get_request(self, plat: Platform) -> ExternalToolRequest:
             f"{self.known_versions}"
         )
 
+    def check_version_constraint(self, version: str) -> None:
+        if not self.constraints or self.constraints.specifier.contains(version):  # type: ignore[attr-defined]
+            # all ok
+            return None
+
+        action = self.options.use_unsupported_version
```

Comment:
You can use the same base error for all 3 types:

> The option `[{self.options_scope}].version` is set to {version}, which is not compatible with what this release of Pants expects: {self.constraints}. Please update the version to a supported value, or consider using a different Pants release if you cannot change the version. 

Then you can append things like, for errors,

> Alternatively, update `[{self.options_scope}].use_unsupported_version` to be `allow` or `warn`.